kpatch-build/kpatch-elf.c

```diff
@@ -726,6 +753,119 @@ void kpatch_reindex_elements(struct kpatch_elf *kelf)
 	}
 }
 
+void kpatch_mark_grouped_sections(struct kpatch_elf *kelf)
+{
+	struct section *groupsec, *sec;
+	struct group_section *gsec;
+	struct symbol *tsym;
+	unsigned int *data, *end, i;
+
+	INIT_LIST_HEAD(&kelf->groupsec);
+
+	list_for_each_entry(groupsec, &kelf->sections, list) {
+		i = 0;
+		if (groupsec->sh.sh_type != SHT_GROUP)
+			continue;
+		/*
+		 * a. Maintain the section index in struct section and struct
+		 *    group_section. This is later used to compare if the reindexed
+		 *    section and group_section matches.
+		 * b. Maintain the secnames, symnames to later update
+		 *    group section members sh_link and sh_info.
+		 */
+		groupsec->groupindex = groupsec->index;
+		data = groupsec->data->d_buf;
+		end = groupsec->data->d_buf + groupsec->data->d_size;
+		data++; /* skip first flag word (e.g. GRP_COMDAT) */
+		gsec = malloc(sizeof(*gsec));
```

Comment:
I may be the only one who resorts to using Valgrind now and then to debug problems in this space, but can we clean up these new allocations.  No need to worry about all the error path scenarios, but at least successful execution will run `kpatch_elf_teardown()` on its way out.

Reply:
Thank you. Will do the cleanup of allocations.